### PR TITLE
US040 - Login System

### PR DESCRIPTION
**Pull request by gerardmclean2**
Added integration migrations, created login functionality along with unit and integration tests

### CODE REVIEW

File: src/main/java/org/example/Exceptions/Entity.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package org.example.Exceptions;
+
+public enum Entity {
```
Comment (shaunganley) [inline src/main/java/org/example/Exceptions/Entity.java]: should this be a UserRole enum?

File: src/main/resources/db/migration/V3__create_user_db.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE `User`(
+Username varchar(64) NOT NULL,
+Password char(60) NOT NULL,
+RoleID TINYINT NOT NULL,
+PRIMARY KEY (Username),
+FOREIGN KEY (RoleID) REFERENCES Role(RoleID)
+);
+
+INSERT INTO `User`(Username, Password, RoleID) VALUES("admin", "$2a$10$0fVhElli6zf/AkCt03J2W.0LoQRnl9CR/iY2Pu38LymHqOspE0bha", 1);
+INSERT INTO `User`(Username, Password, RoleID) VALUES("user", "$2a$10$aGs2jdxL1nyvAPj9D8Z8IOuLaEF.Fr5L9LVp9C33YFgX6fDN1RBK6", 2);
```
Comment (shaunganley) [inline src/main/resources/db/migration/V3__create_user_db.sql:10]: if this migration is running against prod we probably don't want to set the passwords in our code repo

File: src/test/java/org/example/integration/AuthIntegrationTests.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+@ExtendWith(DropwizardExtensionsSupport.class)
+public class AuthIntegrationTests {
+    public static final DropwizardAppExtension<TestConfiguration> APP =
+            new DropwizardAppExtension<>(TestApplication.class);
+
+    @Test
+    void login_shouldLogin_whenCorrectCredentials(){
+        Client client = APP.client();
+        LoginRequest login = new LoginRequest("admin", "admin");
```
Comment (shaunganley) [inline src/test/java/org/example/integration/AuthIntegrationTests.java]: again, shouldn't have valid login creds in code, you can pass these in as env vars

File: src/test/java/org/example/validator/AuthValidatorTests.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package org.example.validator;
+
+public class AuthValidatorTests {
```
Comment (shaunganley) [inline src/test/java/org/example/validator/AuthValidatorTests.java]: not needed?

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/test/java/org/example/integration/AuthIntegrationTests.java (hunk 3 above)
Comment (gerardmclean2) [reply src/test/java/org/example/integration/AuthIntegrationTests.java]: Created env vars for user and pass

File: src/test/java/org/example/validator/AuthValidatorTests.java (hunk 4 above)
Comment (gerardmclean2) [reply src/test/java/org/example/validator/AuthValidatorTests.java]: Deleted file

File: src/main/java/org/example/Exceptions/Entity.java (hunk 1 above)
Comment (gerardmclean2) [reply src/main/java/org/example/Exceptions/Entity.java]: Deleted file as the enum ended up not being used.

**Review by JakubRS7 — APPROVED** [review]

**Review by shaunganley — APPROVED** [review]

**Pull request merged**